framework/src/st_things/things_stack/utils/things_ping.c

```diff
@@ -964,6 +890,7 @@ static bool set_interval(things_ping_s *ping, int64_t *intervals, int int_ea, pi
 	if (get_mask(ping, confirmState) == true && get_mask(ping, PING_ST_TIMEOUT) == false) {
 		min_interval = find_minimum_from_intervals(intervals, int_ea);
 		memset(ping->interval, 0, sizeof(int64_t) * MAX_INTERVAL_EA);
+		int i = 0;
```

Comment:
can variable i be moved inside for() ..  ?

for(int i = 0;...)

Reply:
No. We do not declare data in for statement.
```
int i = 0;
for (i = 0; ....)
```
this is right.
Because it is not supported by some compilers.